exaudfclient/base/script_options_parser/ctpg/script_option_lines_ctpg.cc

```diff
@@ -0,0 +1,195 @@
+        options(options, option_element)
+            >= [](auto&& ob, auto&& e) { return add_option(std::move(e), std::move(ob)); },
+        option_element(start_option_tag, alpha_numeric, whitespaces, option_value, end_option_tag)
+            >= [](auto st, auto ok, skip, auto ov, auto e) { return to_option(ok.get_value(), ov, st.get_sp(), e.get_sp()); },
+        option_element(whitespaces, start_option_tag, alpha_numeric, whitespaces, option_value, end_option_tag)
+            >= [](skip, auto st, auto ok, skip, auto ov, auto e) { return to_option(ok.get_value(), ov, st.get_sp(), e.get_sp()); },
+        option_value(alpha_numeric)
+            >= [](auto o) { return std::string(o.get_value()); },
+        option_value(option_char)
+            >= [](auto o) { return std::string(o.get_value()); },
+        option_value(whitespaces)
+            >= [](auto o) { return std::string(o.get_value()); },
+        option_value(semicolon_escape)
+            >= [](auto o) { return std::string(";"); },
+        option_value(option_value, option_char)
+            >= [](auto&& ov, auto v) { return std::move(ov.append(v.get_value())); },
+        option_value(option_value, semicolon_escape)
+            >= [](auto&& ov, auto v) { return std::move(ov.append(";")); },
+        option_value(option_value, start_option_tag)
+            >= [](auto&& ov, auto v) { return std::move(ov.append("%")); },
+        option_value(option_value, alpha_numeric)
+            >= [](auto&& ov, auto v) { return std::move(ov.append(v.get_value())); },
+        option_value(option_value, whitespaces)
+            >= [](auto&& ov, auto v) { return std::move(ov.append(v.get_value())); },
+        rest(alpha_numeric)
+            >= [](auto r) { return 0;},
+        rest(whitespaces)
+            >= [](auto r) { return 0;},
+        rest(semicolon_escape)
+            >= [](auto r) { return 0;},
+        rest(end_option_tag)
+            >= [](auto r) { return 0;},
+        rest(option_char)
+            >= [](auto r) { return 0;},
+        rest(rest, alpha_numeric)
+            >= [](auto r, skip) { return 0;},
+        rest(rest, option_char)
+            >= [](auto r, skip) { return 0;},
+        rest(rest, whitespaces)
+            >= [](auto r, skip) { return 0;},
+        rest(rest, end_option_tag)
+            >= [](auto r, skip) { return 0;},
+        rest(rest, start_option_tag)
+            >= [](auto r, skip) { return 0;}
+    )
+);
+
+void parse(const std::string& code, options_type& result, std::function<void(const char*)> throwException) {
+    std::stringstream error_buffer;
+    auto res = option_parser.parse(
+        parse_options{}.set_skip_whitespace(false),
+        buffers::string_buffer(code.c_str()),
+        error_buffer);
+    if (res.has_value())
+    {
+        result = res.value();
+    }
+    else
+    {
+        std::stringstream ss;
+        ss << "Error parsing script options: " << error_buffer.str();
+        throwException(ss.str().c_str());
```

Comment:
I would prefer to keep it as is.
`throwException` expects a const char*. Inlining means I need to construct a temp std::string and then pass the internal pointer. I would like to avoid any possible source of memory issue here.